### PR TITLE
Added example for :class:`~.Sphere`

### PR DESCRIPTION
**Pull request by jojo96**
## Changelog / Overview

Added documentation for the sphere class. Added an example since there were no existing examples for the sphere class

## Motivation

Added example for new users
## Explanation for Changes

Makes it more readable as the examples are useful to the newer users

[Documentation Reference](https://manimce--1627.org.readthedocs.build/en/1627/reference/manim.mobject.three_dimensions.Sphere.html)

## Checklist
- [x] I have read the [Contributing Guidelines](https://docs.manim.community/en/latest/contributing.html)
- [ ] I have written a descriptive PR title (see top of PR template for examples)
- [x] I have written a changelog entry for the PR or deem it unnecessary
- [ ] My new functions/classes either have a docstring or are private
- [ ] My new functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [x] My new documentation builds, looks correctly formatted, and adds no additional build warnings

## Reviewer Checklist
- [ ] The PR title is descriptive enough
- [x] The PR is labeled correctly
- [ ] The changelog entry is completed if necessary
- [ ] Newly added functions/classes either have a docstring or are private
- [ ] Newly added functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [x] Newly added documentation builds, looks correctly formatted, and adds no additional build warnings

### CODE REVIEW

**Comment by Darylgolden** [comment]
[Courtesy link](https://manimce--1627.org.readthedocs.build/en/1627/reference/manim.mobject.three_dimensions.Sphere.html?highlight=sphere#manim.mobject.three_dimensions.Sphere)

**Comment by Darylgolden** [comment]
Hi @jojo96, thank you for your contribution! Personally I don't like the look of the spheres overlapping, could you change that perhaps? It would be nice to have different colors as well.

**Comment by kolibril13** [comment]
Thanks for the contribution!
Two suggestions:
A) I also think it is better when the spheres do not overlap.
B) I would name them sphere1, sphere2, sphere3

**Comment by jojo96** [comment]
> Hi @jojo96, thank you for your contribution! Personally I don't like the look of the spheres overlapping, could you change that perhaps? It would be nice to have different colors as well.


> Thanks for the contribution!
> Two suggestions:
> A) I also think it is better when the spheres do not overlap.
> B) I would name them sphere1, sphere2, sphere3

Hi I have created spheres with better resolutions and changed colors as well. Also, implemented A and B as suggested. Added these to patch-2. 

Do I need to create a new pull request? @kolibril13

**Comment by kolibril13** [comment]
Great! 
No, you don't have to recreate a pr.
You just have to wait until this test is passing,
image
and the new example will be available here:
https://manimce--1627.org.readthedocs.build/en/1627/reference/manim.mobject.three_dimensions.Sphere.html?highlight=sphere#manim.mobject.three_dimensions.Sphere

File: manim/mobject/three_dimensions.py [hunk 1]
```diff
@@ -134,6 +134,30 @@ def set_fill_by_checkerboard(self, *colors, opacity=None):
...
+    .. manim:: ExampleSphere
+        :save_last_frame:
+
+        class ExampleSphere(ThreeDScene):
+            def construct(self):
+                self.set_camera_orientation(phi = PI/6, theta = PI/6)
+                sphere1 = Sphere(center = (3,0,0),radius = 1,resolution = (30, 40),u_min = 0.001,u_max = PI - 0.001,v_min = 0,v_max = TAU)
+                sphere1.set_color(RED)
+                self.add(sphere1)
+                sphere2 = Sphere(center=(0,1,0),radius=2,resolution=(40, 50))
```
Comment (hydrobeam) [inline manim/mobject/three_dimensions.py:150]: What's the reasoning behind specifying `center` and then using `move_to` ? (same goes for sphere3)

File: manim/mobject/three_dimensions.py [hunk 2]
```diff
@@ -134,6 +134,30 @@ def set_fill_by_checkerboard(self, *colors, opacity=None):
...
+                sphere2.move_to((-1,-3,0))
+                sphere2.set_color(GREEN)
+                self.add(sphere2)
+                sphere3 = Sphere(center=(0,1,0),radius=2,resolution=(40, 50))
```
Comment (hydrobeam) [inline manim/mobject/three_dimensions.py:154]: All these examples are pretty high res. I would recommend showing what a low res sphere would look like too. This is a sample value I toyed around with. 
```suggestion
                sphere3 = Sphere(center=(0,1,0),radius=2,resolution=(14, 14))
```

**Review by hydrobeam — CHANGES_REQUESTED** [review]
Styling + more variety in resolutions

File: manim/mobject/three_dimensions.py (hunk 2 above)
Comment (hydrobeam) [reply manim/mobject/three_dimensions.py]: This is a bit of an afterthought but: dropping the resolution of sphere1 to `(20, 30)` and sphere2 to `(35, 40)` would hardly affect their look and save a couple seconds in rendering time.

File: manim/mobject/three_dimensions.py (hunk 1 above)
Comment (jojo96) [reply manim/mobject/three_dimensions.py]: Removed the move_to functions.

File: manim/mobject/three_dimensions.py [hunk 3]
```diff
@@ -134,6 +134,30 @@ def set_fill_by_checkerboard(self, *colors, opacity=None):
 
 
 class Sphere(ParametricSurface):
+    """A sphere
```
[the remark this answers is not in the document]
Comment (naveen521kk) [reply manim/mobject/three_dimensions.py:137]: I think this docstring too obvious. Either add few more lines or remove this.

File: manim/mobject/three_dimensions.py [hunk 4]
```diff
@@ -134,6 +134,30 @@ def set_fill_by_checkerboard(self, *colors, opacity=None):
...
+                self.add(sphere1)
+                sphere2 = Sphere(center=(0,1,0),radius=2,resolution=(40, 50))
+                sphere2.move_to((-1,-3,0))
+                sphere2.set_color(GREEN)
+                self.add(sphere2)
+                sphere3 = Sphere(center=(0,1,0),radius=2,resolution=(40, 50))
+                sphere3.move_to((-1,2,0))
+                sphere3.set_color(BLUE)
+                self.add(sphere3)
+
```
[the remark this answers is not in the document]
Comment (naveen521kk) [reply manim/mobject/three_dimensions.py]: FYI, using the bot should do this automatically.

File: manim/mobject/three_dimensions.py (hunk 2 above)
Comment (jojo96) [reply manim/mobject/three_dimensions.py]: @naveen521kk removed sphere from top line. Also copied the documentation from the bot. 

And also there is no hackathon label added to this thread.

File: manim/mobject/three_dimensions.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (behackl) [reply manim/mobject/three_dimensions.py]: > I think this docstring too obvious. Either add few more lines or remove this.

No, that's unfortunately bad advice. Every docstring **needs** to start with a one-line summary. If it is just "A Sphere.", then that's also just fine. A longer version ("A Mobject representing a threedimensional sphere.") would also be fine.

File: manim/mobject/three_dimensions.py [hunk 5]
```diff
@@ -134,6 +134,28 @@ def set_fill_by_checkerboard(self, *colors, opacity=None):
 
 
 class Sphere(ParametricSurface):
+    """
```
Comment (behackl) [inline manim/mobject/three_dimensions.py:137]: Just as mentioned in the other comment: every docstring **needs** to start with a one-line summary. Please add back something along the lines of

```suggestion
    """A mobject representing a three-dimensional sphere.
```

(feel free to commit your own suggestion instead.)

**Review by behackl — CHANGES_REQUESTED** [review]
Left one change request. @hydrobeam's comment regarding the resolution (which has a rather large impact on the render time in our documentation) also seems to have not been addressed, could you please take a look again?

Thank you!

**Comment by jojo96** [comment]
> Left one change request. @hydrobeam's comment regarding the resolution (which has a rather large impact on the render time in our documentation) also seems to have not been addressed, could you please take a look again?
> 
> Thank you!

@behackl reduced the resolution of the spheres to (20,20), (18,18), (16,16). Reducing the resolution further does not render good looking spheres.

**Review by behackl — APPROVED** [review]
I'm happy with this; slightly higher resolution would probably be fine as well. Let's see what @hydrobeam has to say.

File: manim/mobject/three_dimensions.py [hunk 6]
```diff
@@ -134,6 +134,28 @@ def set_fill_by_checkerboard(self, *colors, opacity=None):
...
+                self.set_camera_orientation(phi=PI / 6, theta=PI / 6)
+                sphere1 = Sphere(center=(3, 0, 0), radius=1, resolution=(20, 20), u_min=0.001, u_max=PI - 0.001, v_min=0, v_max=TAU)
+                sphere1.set_color(RED)
+                self.add(sphere1)
+                sphere2 = Sphere(center=(-1, -3, 0), radius=2, resolution=(18, 18))
+                sphere2.set_color(GREEN)
+                self.add(sphere2)
+                sphere3 = Sphere(center=(-1, 2, 0), radius=2, resolution=(16, 16))
+                sphere3.set_color(BLUE)
+                self.add(sphere3)
```
Comment (hydrobeam) [inline manim/mobject/three_dimensions.py:156]:
```suggestion
        class ExampleSphere(ThreeDScene):
            def construct(self):
                self.set_camera_orientation(phi=PI / 6, theta=PI / 6)
                sphere1 = Sphere(
                    center=(3, 0, 0),
                    radius=1,
                    resolution=(20, 20),
                    u_min=0.001,
                    u_max=PI - 0.001,
                    v_min=0,
                    v_max=TAU,
                )
                sphere1.set_color(RED)
                self.add(sphere1)
                sphere2 = Sphere(center=(-1, -3, 0), radius=2, resolution=(18, 18))
                sphere2.set_color(GREEN)
                self.add(sphere2)
                sphere3 = Sphere(center=(-1, 2, 0), radius=2, resolution=(16, 16))
                sphere3.set_color(BLUE)
                self.add(sphere3)
```

**Review by hydrobeam — APPROVED** [review]
Looks good to me too. Just apply black formatting here so that you don't have to scroll horizontally to see all the code.

**Review by Darylgolden — APPROVED** [review]
LGTM

**Pull request merged**